judgels-frontends/raphael/src/routes/uriel/contests/single/contestants/ContestContestantsTable/ContestContestantsTable.tsx

```diff
@@ -22,11 +23,16 @@ export class ContestContestantsTable extends React.PureComponent<ContestContesta
     );
   }
 
+  private isVirtualContest = () => {
```

Comment:
This will be called multiple times unnecessarily.

We can pass the result as parameter to `renderHeader()` and `renderRows()`.